docs/source/conf.py

```diff
@@ -109,6 +127,102 @@
 # do not prepend module name to functions
 add_module_names = False
 
+
+with open('../../numpyro/version.py') as f:
+    for line in f.readlines():
+        if line.startswith("__version__ = "):
+            break
+    pip_version = line.rstrip().split(" = ")[-1].strip("'").strip('"')
+    # resolve the issue: tags for versions before 0.3.0 have the prefix 'v'.
+    tag = "v" + pip_version if version[:3] < "0.3" else pip_version
+
+
+# This is processed by Jinja2 and inserted before each notebook
+nbsphinx_prolog = r"""
+{% set docname = 'notebooks/source/' + env.doc2path(env.docname, base=None).split('/')[-1] %}
+.. raw:: html
+
+    <div class="admonition note">
+      Interactive online version:
+      <span style="white-space: nowrap;">
+        <a href="https://colab.research.google.com/github/pyro-ppl/numpyro/blob/{tag}/{{ docname }}">
+          <img alt="Open In Colab" src="https://colab.research.google.com/assets/colab-badge.svg"
+            style="vertical-align:text-bottom">
+        </a>
+      </span>
+    </div>
+""".replace(r"{tag}", tag)
+
+
+# -- Copy README files
+
+# replace "# NumPyro" by "# Getting Started with NumPyro"
+with open('../../README.md', 'rt') as f:
+    lines = f.readlines()
+    for i, line in enumerate(lines):
+        if "# NumPyro" == line.rstrip():
+            break
+    lines = lines[i:]
+    lines[0] = "# Getting Started with NumPyro\n"
+    text = "\n".join(lines)
+
+with open('getting_started.rst', 'wt') as f:
+    f.write(nbsphinx.markdown2rst(text))
+
+
+# -- Copy notebook files
+
+if not os.path.exists('tutorials'):
+    os.makedirs('tutorials')
+
+# remove files that are updated or not available in notebooks/source
```

Comment:
I did not exactly follow this part - why do we need to copy these files from `notebooks/source` directory? Also will these files persist after we have built the documentation?

Reply:
Yes, these files will persist so I have to add some entries to [.gitignore](https://github.com/pyro-ppl/numpyro/pull/791/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947R6) to skip committing those files. This works but not a good solution. I don't think `sphinx` can provide an easy way to specify files outside of `source` directory but let me check it again.

Reply:
I see. If that doesn't work, we can temporarily create these directories but clean it up before exiting.

Reply:
I just took a look at gpytorch and they did the same thing: copy [examples folder](https://github.com/cornellius-gp/gpytorch/blob/8f9b44fc57dbb0a13b568946f07a37e9332f92c4/docs/source/conf.py#L107) and add an entry to `.gitignore`. :D

> clean it up before exiting

If we clean those up then no need to add those md5hash codes, so I'll remove those codes too. I think we can extend the `make docs` command for this. Let me see.

Reply:
@neerajprad I added `git clean -dfx docs/source/` to `Makefile`, which does
```
Removing docs/source/examples/
Removing docs/source/getting_started.rst
Removing docs/source/tutorials/
```
at the end of `make docs`.

Reply:
Sounds good. Minor nit, but I think this will also delete any other new files and leave untracked files if the whole thing failed due to other reasons. Can we just wrap the whole operation in a try, finally block and only delete the directories created?

```python
try:
  # create directories for sphinx
finally:
  # delete these directories (check if they exist first)
```

Reply:
I just added `-` in front of sphinx-build command so `git clean` will be executed even if the build fails for some reason. I think it is better because try/catch can't deal with `examples` folder (which is created by the `sphinx-gallery` extension).